Control/Concurrent/Async.hs

```diff
@@ -478,6 +479,7 @@ waitAnyCatch = atomically . waitAnyCatchSTM
 --
 -- @since 2.1.0
 waitAnyCatchSTM :: [Async a] -> STM (Async a, Either SomeException a)
+waitAnyCatchSTM [] = error "waitAnyCatchSTM: invalid argument: input list must be non-empty"
```

Comment:
1. please throw an exception, don't use `error`
2. keep lines under 80 columns

Reply:
> 1. please throw an exception, don't use `error`

In the associated issue you said using `ErrorCall` is fine. And calling `error` is the same as throwing `ErrorCall` according to https://www.stackage.org/haddock/lts-19.5/base-4.15.1.0/Control-Exception.html#t:ErrorCall. What am I missing?

Reply:
`error` is not the same as `throwIO`: https://hackage.haskell.org/package/base-4.16.1.0/docs/Control-Exception.html#v:throwIO